### PR TITLE
Delete context when stopped

### PR DESCRIPTION
**Pull request by prezha**
fixes: #9410 

as stated in the original issue: if `keep-context-active` is not set to `true`, the context for the stopped cluster should be deleted (instead of just unset as the current context) to avoid confusion

**example**:

```
❯ kubectl config get-contexts
CURRENT   NAME                          CLUSTER                    AUTHINFO                                                       NAMESPACE
          minikube                      minikube                   minikube
*         p2                            p2                         p2
          p3                            p3                         p3
```
```
❯ minikube stop -p p2
✋  Stopping node "p2"  ...
🛑  Powering off "p2" via SSH ...
🛑  1 nodes stopped.
```
**current behaviour**:
```
❯ kubectl config get-contexts
CURRENT   NAME                          CLUSTER                    AUTHINFO                                                       NAMESPACE
          minikube                      minikube                   minikube
          p2                            p2                         p2
          p3                            p3                         p3
```
**new behaviour**:
```
❯ kubectl config get-contexts
CURRENT   NAME                          CLUSTER                    AUTHINFO                                                       NAMESPACE
          minikube                      minikube                   minikube
          p3                            p3                         p3
```
```
❯ minikube start -p p2
😄  [p2] minikube v1.13.1 on "Opensuse-Tumbleweed"
✨  Using the docker driver based on existing profile
❗  docker is currently using the btrfs storage driver, consider switching to overlay2 for better performance
👍  Starting control plane node p2 in cluster p2
🔄  Restarting existing docker container for "p2" ...
🐳  Preparing Kubernetes v1.19.2 on Docker 19.03.8 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "p2" by default
```
```
❯ kubectl config get-contexts
CURRENT   NAME                          CLUSTER                    AUTHINFO                                                       NAMESPACE
          minikube                      minikube                   minikube
*         p2                            p2                         p2
          p3                            p3                         p3
```

### CODE REVIEW

File: cmd/minikube/cmd/stop.go [hunk 1]
```diff
@@ -115,8 +115,8 @@ func stopProfile(profile string) int {
 	}
 
 	if !keepActive {
-		if err := kubeconfig.UnsetCurrentContext(profile, kubeconfig.PathFromEnv()); err != nil {
-			exit.Error(reason.HostKubeconfigUnset, "update config", err)
+		if err := kubeconfig.DeleteContext(profile, kubeconfig.PathFromEnv()); err != nil {
+			exit.Error(reason.HostKubeconfigUpdate, "update config", err)
```
Comment (medyagh) [inline cmd/minikube/cmd/stop.go:119]: lets create a new Reason Exit here 
https://github.com/medyagh/minikube/blob/82a5189b934ea02d127be85dbdba50dfa4282984/pkg/minikube/reason/reason.go#L207

```
	HostKubeconfigDelete = Kind{ID: "HOST_KUBECONFIG_DELETE", ExitCode: ExHostConfig}
```

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/stop.go (hunk 1 above)
Comment (prezha) [reply cmd/minikube/cmd/stop.go:119]: @medyagh you are most welcome!
that did cross my mind, but `HostKubeconfigDelete` sounded a bit wrong (we are not going to delete kubeconfig), while `HostKubeconfigUpdate` not so bad as we are effectively just updating the kubeconfig
on the other hand, maybe:
```
	HostKubeconfigContextDelete = Kind{ID: "HOST_KUBECONFIG_CONTEXT_DELETE", ExitCode: ExHostConfig}
```
would be more appropriate?
what do you think?

Comment (medyagh) [reply cmd/minikube/cmd/stop.go:119]: that is actually good thinking ! I like that

how about something shorter but unique

"HostKubeconfigDeleteCtx"

the goal is if users collect these exit codes as metrics, they should find out what was the cause of the problem in a unique word

Comment (prezha) [reply cmd/minikube/cmd/stop.go:119]: yes, that sounds just about right :)
the update is on its way...
thanks @medyagh !

**Review by medyagh — APPROVED** [review]
thanks

**Pull request merged**